ingestion/src/metadata/ingestion/source/trino.py

```diff
@@ -27,16 +27,17 @@ class TrinoConfig(SQLConnectionConfig):
     schema_name: Optional[str]
 
     def get_connection_url(self):
-        url = f"{self.scheme}://"
-        if self.username:
-            url += f"{quote_plus(self.username)}"
-            if self.password:
-                url += f":{quote_plus(self.password)}"
-        url += f"{self.host_port}"
+        url = f"{self.scheme}://{self.host_port}"
         if self.catalog:
             url += f"/{quote_plus(self.catalog)}"
             if self.schema_name:
                 url += f"/{quote_plus(self.schema_name)}"
+        if self.username:
+            url += f"?user={quote_plus(self.username)}"
+            if self.password:
+                url += f"&password={quote_plus(self.password)}"
+        if self.options:
+            url = self.get_options(options=self.options, url=url)
```

Comment:
May I suggest using [urlencode](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode) instead here? It is more idiomatic python code.

Reply:
I think we rant into issues with url encode but we can give it a try